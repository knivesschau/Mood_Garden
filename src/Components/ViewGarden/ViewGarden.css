/* mobile first styles, min 320px - max 414px */

#garden-header {
    text-align: center;
}

#garden-header {
    font-size: 1.90em;
}

ul {
    margin-left: auto;
    margin-right: auto;
    padding-left: .25em;
}

li {
    text-align: center;
    list-style-type: none; 
    background-color: rgb(255,173,187, .90);
    box-shadow: inset -11px -5px 5px 0px #A8739B;
    border-radius: 1px transparent white;
    border-radius: 20em; 
}

#welcome {
    text-align: center;
    border: 1px solid black;
    background-color: rgb(203, 211, 167, .90);
    box-shadow: inset -4px 0px 5px 2px #6E7C51; 
    border-radius: 2em;
    font-size: .90em;
    margin-top: -1em;
    margin-bottom: 1.5em;
    padding: .5em;
}

@media only screen and (min-device-width: 414px) { 
    #edit-header {
        font-size: 2em;
    }
    
    #garden-header {
        font-size: 2.5em;
        margin-top: .50em;
    }

    #welcome {
        font-size: 1em;
        padding: .75em;
        margin-left: 1em;
        margin-right: 1em;
    }
}

/* desktop and tablet styles. query order is descending from larger desktops - tablets due to max-width constraints */

@media only screen and (min-device-width: 1700px) and (max-width: 2200px) {
    ul {
        margin-left: 25em;
        margin-right: 25em;
    }

    #garden-header {
        font-size: 3.5em;
    }

    #welcome {
        font-size: 1.75em;
        margin-left: 10em;
        margin-right: 10em;
    }
}

@media only screen and (min-device-width: 1200px) and (max-width: 1699px) {
    ul {
        margin-left: 10em;
        margin-right: 10em;
    }

    #garden-header {
        font-size: 3.25em;
    }

    #welcome {
        font-size: 1.5em;
        margin-left: 7.5em;
        margin-right: 7.5em;
    }
} 

@media only screen and (min-device-width: 768px) and (max-width: 1199px) {
    ul {
        margin-left: 3em;
        margin-right: 3em;
    }

    #garden-header {
        font-size: 3em;
        margin-bottom: .50em;
    }

    #welcome {
        font-size: 1.25em;
        padding: 1.5em;
        margin-top: .5em;
        margin-left: 5em;
        margin-right: 5em;
    }
}