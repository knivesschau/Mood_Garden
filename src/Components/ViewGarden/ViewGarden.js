import React, {Component} from 'react';
import moodGardenContext from '../../moodGardenContext';
import RoseInfo from '../RoseInfo/RoseInfo';
import './ViewGarden.css';

export default class ViewGarden extends Component {
    static contextType = moodGardenContext;

    render() {
        const {roses=[]} = this.context;

        return (
            <section className="ViewGarden">
                
                <h1 id="garden-header">Your Mood Garden</h1>
            
                <ul>
                    {roses.map(rose => 
                        <li key={rose.id}> 
                            <RoseInfo
                                id={rose.id}
                                color={rose.color}
                                entry_date={rose.entry_date}
                                // rose={rose.rose}
                                // thorn={rose.thorn}
                                // bud={rose.bud}
                            />
                        </li>
                    )}
                </ul>

            </section>
        )
    }
}


