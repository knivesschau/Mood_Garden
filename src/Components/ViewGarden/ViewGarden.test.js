import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewGarden from './ViewGarden';

describe ('ViewGarden Component', () => {
    it ('renders the complete ViewGarden Component and all nested Components', () => {
        const wrapper = shallow(<ViewGarden/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})