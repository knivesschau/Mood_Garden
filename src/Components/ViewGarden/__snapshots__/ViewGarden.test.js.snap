// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewGarden Component renders the complete ViewGarden Component and all nested Components 1`] = `
<section
  className="ViewGarden"
>
  <nav
    className="Garden_Nav"
    role="navigation"
  >
    <RoseNav />
  </nav>
  <h1
    id="garden-header"
  >
    Your Mood Garden
  </h1>
  <ul />
</section>
`;
